components/style/matching.rs

```diff
@@ -769,22 +769,27 @@ pub trait MatchMethods : TElement {
                 //
```

Comment:
I think it would be an improvement - right now compute_restyle_damage and existing_style_for_restyle_damage both take an Option for old_style, and need to handle that case, and then the caller just ignores the result when old_style was none.

I guess it's slightly complicated by the pseudo-handling callsite though. In any case, we can do it later, so no need to worry about it for now.